qemu/tests/virtio_driver_sign_check.py

```diff
@@ -1,87 +1,153 @@
+
+        Collect all tuples in virtio_drivers list to be used signtool test.
+
+        :param filterd_cat_file:each filterd suffix '.cat' file
+        """
+        pattern = r'\w:\\(.+?)\\(.+?)\\(.+?)[\.|\\]'
+        match_ret = re.match(pattern, filterd_cat_file)
+        driver_dir = match_ret.group()
+
+        driver_sys = filter(lambda x: driver_dir in x, drivers['.sys'])
+        driver_inf = filter(lambda x: driver_dir in x, drivers['.inf'])
+        driver_wdf = filter(lambda x: driver_dir in x, drivers['Wdf'])
+
+        cat_sys = map(lambda x: (filterd_cat_file, x), driver_sys)
+        cat_inf = map(lambda x: (filterd_cat_file, x), driver_inf)
+        cat_wdf = map(lambda x: (filterd_cat_file, x), driver_wdf)
+        cat_parm = cat_sys + cat_inf + cat_wdf
+        virtio_drivers.extend(cat_parm)
+
     try:
-        for drive in drive_list.split():
-            for type in ['.cat', '.sys']:
+        error_context.context("Running SignTool check test in guest...",
+                              logging.info)
+        key = "VolumeName like 'virtio-win%'"
+        disk_letter = utils_misc.get_win_disk_vol(session, condition=key) + ":"
+        drive_volume_list.append(disk_letter)
+
+        for drive in drive_volume_list:
+            for type in ['.cat', '.sys', '.inf', 'Wdf']:
                 driver = session.cmd_output(list_files_cmd % (drive, type)).\
                     splitlines()[1:-1]
                 drivers[type] = driver
-            files = zip(drivers['.cat'], drivers['.sys'])
-            for cat, sys in files:
-                cmd = signtool_cmd % (cat, sys)
-                s, result = session.cmd_status_output(cmd)
-                if s:
-                    msg = "Fail command: %s. Output: %s" % (cmd, result)
-                    raise error.TestFail(msg)
+
+            cat_list = filter(filter_all_tested_cat_files, drivers['.cat'])
+            map(match_cat_and_verified_file, cat_list)
+
+            for driver_file in virtio_drivers:
+                test_cmd = signtool_cmd % (driver_file[0], driver_file[1])
+                test_cmd = utils_misc.set_winutils_letter(session, test_cmd)
+                status, result = session.cmd_status_output(test_cmd)
+                if status:
+                    msg = "Fail command: %s. Output: %s" % (test_cmd, result)
+                    error_context.context("Fail command: %s. Output: %s" %
+                                          (test_cmd, result), logging.info)
                 results_all += result
                 re_suc = "Number of files successfully Verified: ([0-9]*)"
                 try:
-                    suc_num = re.findall(re_suc, result)[0]
+                    suc_num = re.findall(re_suc, result, re.M)[0]
                 except IndexError:
                     msg = "Fail to get Number of files successfully Verified"
-                    raise error.TestFail(msg)
+                    test.fail(msg)
```

Comment:
If you don't expect job be interrupted by one file signcheck, I think you should not use test.fail() here.